data/blog/improving-developer-experience-from-unreadable-css-selectors-to-clear-component-names.md

```diff
@@ -0,0 +1,66 @@
+---
+title: 'Improving Developer Experience: From Unreadable CSS Selectors to Clear Component Names'
+date: '2024-05-30'
+tags: ['web', 'react']
+draft: false
+summary: 'How to transform unreadable CSS selectors to React component names.'
+images: []
+layout: PostLayout
+authors: ['catherinelee']
+---
+
+Reading our own code is much easier than the output of a bundler or transpiler. The same goes for React developers reading CSS selectors generated from their CSS library of choice. CSS selectors are used all across Sentry to show clicks on components. This can provide useful information like which component is the cause of rage and dead clicks, and what component was clicked in a span. However, if you’re using React, these selectors become unreadable in production due to the minification process, so your selectors end up looking like this:
```

Comment:
One might land on this post without knowing what `Sentry` is:

```suggestion
Reading our own code is much easier than the output of a bundler or transpiler. The same goes for React developers reading CSS selectors generated from their CSS library of choice. CSS selectors are used all across [Sentry](https://sentry.io/for/react/) to show clicks on components. This can provide useful information like which component is the cause of rage and dead clicks, and what component was clicked in a span. However, if you’re using React, these selectors become unreadable in production due to the minification process, so your selectors end up looking like this:
```